plugins/modules/nd_federation_member.py

```diff
@@ -0,0 +1,271 @@
+        type: str
+      password:
+        description:
+          - The password for the federation member/cluster.
+        type: str
+      login_domain:
+        description:
+          - The login domain ame to use for the federation member/cluster.
+          - Default value is set to DefaultAuth.
+        type: str
+        default: "DefaultAuth"
+  state:
+    description:
+      - The state of the cluster configuration.
+    type: str
+    default: present
+    choices: [ absent, present, query ]
+extends_documentation_fragment: cisco.nd.modules
+notes:
+- The M(cisco.aci.nd_federation) module can be used for this.
+"""
+
+EXAMPLES = r"""
+- name: Setup multi-cluster configuration
+  cisco.nd.nd_federation_member:
+    host: nd
+    username: admin
+    password: SomeSecretPassword
+    clusters:
+      - hostname: 172.37.20.15
+        username: admin
+        password: password
+        login_domain: default
+    state: present
+  delegate_to: localhost
+
+- name: Query all federation members
+  cisco.nd.nd_federation_member:
+    host: nd
+    username: admin
+    password: SomeSecretPassword
+    state: query
+  delegate_to: localhost
+
+- name: Query a federation member
+  cisco.nd.nd_federation_member:
+    host: nd
+    username: admin
+    password: SomeSecretPassword
+    clusters:
+      - hostname: 172.37.20.15
+    state: query
+  delegate_to: localhost
+
+- name: Remove a federation member
+  cisco.nd.nd_federation_member:
+    host: nd
+    username: admin
+    password: SomeSecretPassword
+    state: absent
+  delegate_to: localhost
+"""
+
+RETURN = r"""
+"""
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.cisco.nd.plugins.module_utils.nd import NDModule, nd_argument_spec
+import base64
+
+
+def main():
+    argument_spec = nd_argument_spec()
+    argument_spec.update(
+        clusters=dict(
+            type="list",
+            elements="dict",
+            options=dict(
+                hostname=dict(type="str", required=True, aliases=["cluster_ip", "ip_address", "federation_member"]),
+                username=dict(type="str"),
+                password=dict(type="str", no_log=True),
+                login_domain=dict(type="str", default="DefaultAuth"),
+            ),
+        ),
+        state=dict(type="str", default="present", choices=["absent", "present", "query"]),
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        required_if=[
+            ["state", "present", ["clusters"]],
+        ],
+    )
+
+    nd = NDModule(module)
+
+    clusters = nd.params.get("clusters")
+    state = nd.params.get("state")
+
+    # validate parameters
+    if clusters:
+        for cluster in clusters:
+            if state == "present":
+                if not (cluster.get("username") and cluster.get("password")):
+                    nd.fail_json(msg="'username' and 'password' are required when state is present.")
```

Comment:
We could improve on the nesting and readability here I think:

```suggestion
    clusters = nd.params.get("clusters", [])
    state = nd.params.get("state")

    # validate parameters
    for cluster in clusters:
        if state == "present" and not (cluster.get("username") and cluster.get("password")):
            nd.fail_json(msg="'username' and 'password' are required when state is present.")
```

Reply:
changed and then edited 139 -> clusters = nd.params.get("clusters") if nd.params.get("clusters") is not None else []

Reply:
fixed